lib/ranked-model.rb

```diff
@@ -29,11 +29,33 @@ def self.included base
   private
 
   def handle_ranking
+    @ranked_model_ranking_succeeded = true
     self.class.rankers.each do |ranker|
-      ranker.with(self).handle_ranking
+      ranking_failed_reason = catch :ranking_failed do
+        ranker.with(self).handle_ranking
+      end
+      handle_ranking_failure(ranker, ranking_failed_reason)
+    end
+    @ranked_model_ranking_succeeded
+  end
+
+  def handle_ranking_failure(ranker, ranking_failed_reason)
+    return if ranking_failed_reason.nil?
+    errors.add ranker.column, ranking_failed_reason
+    @ranked_model_ranking_succeeded = false
+    abort_handle_ranking
+  end
+
+  if defined?(Rails) && Rails.version >= '5'
+    def abort_handle_ranking
```

Comment:
is a reasonable way to discover the behavior. I think this is fine even if `ActiveSupport.halt_callback_chains_on_return_false` is true